rs/consensus/dkg/src/payload_builder.rs

```diff
@@ -1339,31 +1466,266 @@ mod tests {
+                            }
+                            _ => assert!(&conf.resharing_transcript().is_none()),
                         }
                     }
                 }
             }
         });
     }
+
+    /// Test the generation of vetkeys
+    ///
+    /// 1. Create a subnet with 4 nodes and a genesis summary
+    /// 2. Add registry entry to add a vetkey
+    /// 3. Check that a config ends up in the next summary
+    /// 4. Check that the summary after that has a matching next transcript
+    /// 5. CHeck that the summary after that has matching current and next transcripts
+    #[test]
+    fn test_vet_key_local_transcript_generation() {
+        ic_test_utilities::artifact_pool_config::with_test_pool_config(|pool_config| {
+            // We'll have a DKG summary inside every 5th block.
+            let dkg_interval_length = 4;
+            // Committee are nodes 0, 1, 2, 3.
+            let committee = (0..4).map(node_test_id).collect::<Vec<_>>();
+            let Dependencies {
+                mut pool,
+                registry_data_provider,
+                registry,
+                replica_config,
+                ..
+            } = dependencies_with_subnet_params(
+                pool_config,
+                subnet_test_id(0),
+                vec![(
+                    5,
+                    SubnetRecordBuilder::from(&committee)
+                        .with_dkg_interval_length(dkg_interval_length)
+                        .build(),
+                )],
+            );
+
+            // Get the latest summary block, which is the genesis block
+            let cup = PoolReader::new(&pool).get_highest_catch_up_package();
+            let dkg_block = cup.content.block.as_ref();
+            assert_eq!(
+                dkg_block.context.registry_version,
+                RegistryVersion::from(5),
+                "The latest available version was used for the summary block."
+            );
+            let summary = dkg_block.payload.as_ref().as_summary();
+            let dkg_summary = &summary.dkg;
+
+            // The genesis summary does not have vetkeys enabled
+            let vet_key_ids = get_enabled_vet_keys(
+                replica_config.subnet_id,
+                &*registry,
+                dkg_block.context.registry_version,
+            )
+            .unwrap();
+            assert_eq!(vet_key_ids.len(), 0);
+
+            assert_eq!(dkg_summary.registry_version, RegistryVersion::from(5));
+            assert_eq!(dkg_summary.height, Height::from(0));
+            assert_eq!(
+                cup.get_oldest_registry_version_in_use(),
+                RegistryVersion::from(5)
+            );
+
+            // This summary does not contain a config for the vetkey yet
+            assert_eq!(dkg_summary.configs.len(), 2);
+            assert_eq!(dkg_summary.current_transcripts().len(), 2);
+
+            // Since it is the genesis summary, it also has no current transcripts
```

Comment:
The comment says there are no current transcripts, but 2 lines above you assert that there are 2 current transcripts. So, should "current" in the comment actually be "next"?

Reply:
Good point, I will recheck the comment.

Reply:
It is exactly as you said. The comment should say "next". I changed it